src/slack.coffee

```diff
@@ -214,21 +205,12 @@ class SlackBot extends Adapter
     text
 
   send: (envelope, messages...) ->
-    channel = @client.getChannelGroupOrDMByName envelope.room
-    channel = @client.getChannelGroupOrDMByID(envelope.room) unless channel
-
-    if not channel and @client.getUserByName(envelope.room)
-      user_id = @client.getUserByName(envelope.room).id
-      @client.openDM user_id, =>
-        this.send envelope, messages...
-      return
-
```

Comment:
It looks like if it doesn't find a valid channel in the envelope, it opens a DM with the sending user. The reason and purpose of this logic isn't totally clear to me, but it is a a bit of functionality we probably shouldn't elide.